zcash_client_sqlite/src/wallet.rs

```diff
@@ -676,21 +846,86 @@ pub fn put_tx_data<'a, P>(
 ///
 /// Marking a note spent in this fashion does NOT imply that the
 /// spending transaction has been mined.
-pub fn mark_spent<'a, P>(
+pub fn mark_sapling_note_spent<'a, P>(
     stmts: &mut DataConnStmtCache<'a, P>,
     tx_ref: i64,
     nf: &Nullifier,
 ) -> Result<(), SqliteClientError> {
     stmts
-        .stmt_mark_recived_note_spent
+        .stmt_mark_sapling_note_spent
         .execute(&[tx_ref.to_sql()?, nf.0.to_sql()?])?;
     Ok(())
 }
 
+/// Marks the given UTXO as having been spent.
+#[cfg(feature = "transparent-inputs")]
+pub fn mark_transparent_utxo_spent<'a, P>(
+    stmts: &mut DataConnStmtCache<'a, P>,
+    tx_ref: i64,
+    outpoint: &OutPoint,
+) -> Result<(), SqliteClientError> {
+    let sql_args: &[(&str, &dyn ToSql)] = &[
+        (&":spent_in_tx", &tx_ref),
+        (&":prevout_txid", &outpoint.hash().to_vec()),
+        (&":prevout_idx", &outpoint.n()),
+    ];
+
+    stmts
+        .stmt_mark_transparent_utxo_spent
+        .execute_named(&sql_args)?;
+
+    Ok(())
+}
+
+/// Adds the given received UTXO to the datastore.
+#[cfg(feature = "transparent-inputs")]
+pub fn put_received_transparent_utxo<'a, P: consensus::Parameters>(
+    stmts: &mut DataConnStmtCache<'a, P>,
+    output: &WalletTransparentOutput,
+) -> Result<UtxoId, SqliteClientError> {
+    let sql_args: &[(&str, &dyn ToSql)] = &[
+        (
+            &":address",
+            &output.address().encode(&stmts.wallet_db.params),
+        ),
+        (&":prevout_txid", &output.outpoint.hash().to_vec()),
+        (&":prevout_idx", &output.outpoint.n()),
+        (&":script", &output.txout.script_pubkey.0),
+        (&":value_zat", &i64::from(output.txout.value)),
+        (&":height", &u32::from(output.height)),
+    ];
+
+    stmts
+        .stmt_insert_received_transparent_utxo
+        .execute_named(&sql_args)?;
+
+    Ok(UtxoId(stmts.wallet_db.conn.last_insert_rowid()))
+}
+
+/// Removes all records of UTXOs that were recorded as having been received
+/// at block heights greater than the given height. Used in the case of chain
+/// rollback.
```

Comment:
But... it's not? The rollback code has its own UTXO deletion logic (which is the only deletion logic that is working there). Is this being used directly by the mobile SDKs, and if so, how?

Reply:
cc @pacu

Reply:
yes this method is being used on the SDKs to delete UTXOs when new ones are fetched. This is a workaround to avoid having to do bookkeeping logic. This should be improved probably in another PR. 

https://github.com/zcash/ZcashLightClientKit/blob/ae5b3cfc21b5a3e79865fa93e762f6953bc838ff/ZcashLightClientKit/Block/Processor/FetchUnspentTxOutputsOperation.swift#L85

Reply:
I've updated the documentation and marked this method as deprecated, because we really want to handle this differently but I don't want to address that issue in this PR. Opened https://github.com/zcash/librustzcash/issues/500